File: Behavioral/Iterator/Java/SongIterator.java
// SongIterator.java

import java.util.List;

public class SongIterator implements Iterator<Song> {
  private List<Song> songs;
  private int position = 0;

  public SongIterator(List<Song> songs) {
    this.songs = songs;
  }

  @Override
  public boolean hasNext() {
    return position < songs.size();
  }

  @Override
  public Song next() {
    return hasNext() ? songs.get(position++) : null;
  }
}

File: Behavioral/Iterator/Python/song_iterator.py
# song_iterator.py

from iterator import Iterator

class SongIterator(Iterator):
  def __init__(self, songs):
    self._songs = songs
    self._position = 0

  def has_next(self):
    return self._position < len(self._songs)

  def next(self):
    if self.has_next():
      song = self._songs[self._position]
      self._position += 1
      return song
    return None
